collector/collector.go

```diff
@@ -0,0 +1,360 @@
+package collector
+
+import (
+	"log"
+	"regexp"
+	"time"
+
+	bk "gopkg.in/buildkite/go-buildkite.v2/buildkite"
+)
+
+const (
+	RunningBuildsCount   = "RunningBuildsCount"
+	RunningJobsCount     = "RunningJobsCount"
+	ScheduledBuildsCount = "ScheduledBuildsCount"
+	ScheduledJobsCount   = "ScheduledJobsCount"
+	UnfinishedJobsCount  = "UnfinishedJobsCount"
+	TotalAgentCount      = "TotalAgentCount"
+	BusyAgentCount       = "BusyAgentCount"
+	IdleAgentCount       = "IdleAgentCount"
+)
+
+const recordsPerPage = 100
+
+type Opts struct {
+	OrgSlug    string
+	Historical time.Duration
+	Queue      string
+	Debug      bool
+}
+
+type Collector struct {
+	Opts
+
+	buildService interface {
+		ListByOrg(org string, opt *bk.BuildsListOptions) ([]bk.Build, *bk.Response, error)
+	}
+	agentService interface {
+		List(org string, opt *bk.AgentListOptions) ([]bk.Agent, *bk.Response, error)
+	}
+}
+
+func New(c *bk.Client, opts Opts) *Collector {
+	return &Collector{
+		Opts:         opts,
+		buildService: c.Builds,
```

Comment:
Just noticed this in my local copy (I'm doing a fork to add statsd support): should `agentService` be populated too?